tests/e2e/utils/versions.go

```diff
@@ -0,0 +1,307 @@
+	GetAvagoURL() string
+	GetApp() *application.Avalanche
+	GetLatestAvagoByProtoVersion(app *application.Avalanche, rpcVersion int, url string) (string, error)
+}
+
+// NewVersionMapper returns the default VersionMapper for e2e tests
+func NewVersionMapper() VersionMapper {
+	app := &application.Avalanche{
+		Downloader: application.NewDownloader(),
+		Log:        logging.NoLog{},
+	}
+	return &versionMapper{
+		app: app,
+	}
+}
+
+// versionMapper is the default implementation for version mapping.
+// It downloads compatibility URLs from the actual github endpoints
+type versionMapper struct {
+	app *application.Avalanche
+}
+
+// GetLatestAvagoByProtoVersion returns the latest Avalanchego version which
+// runs with the specified rpcVersion, or an error if it can't be found
+// (or other errors occurred)
+func (m *versionMapper) GetLatestAvagoByProtoVersion(app *application.Avalanche, rpcVersion int, url string) (string, error) {
+	return vm.GetLatestAvalancheGoByProtocolVersion(app, rpcVersion, url)
+}
+
+// GetApp returns the Avalanche application instance
+func (m *versionMapper) GetApp() *application.Avalanche {
+	return m.app
+}
+
+// GetCompatURL returns the compatibility URL for the given VM type
+func (m *versionMapper) GetCompatURL(vmType models.VMType) string {
+	switch vmType {
+	case models.SubnetEvm:
+		return constants.SubnetEVMRPCCompatibilityURL
+	case models.SpacesVM:
+		return constants.SpacesVMRPCCompatibilityURL
+	case models.CustomVM:
+		// TODO: unclear yet what we should return here
+		return ""
+	default:
+		return ""
+	}
+}
+
+// GetAvagoURL returns the compatibility URL for Avalanchego
+func (m *versionMapper) GetAvagoURL() string {
+	return constants.AvalancheGoCompatibilityURL
+}
+
+// GetVersionMapping returns a map of specific VMs resp. Avalanchego e2e context keys
+// to the actual version which corresponds to that key.
+// This allows the e2e test to know what version to download and run.
+// Returns an error if there was a problem reading the URL compatibility json
+// or some other issue.
+func GetVersionMapping(mapper VersionMapper) (map[string]string, error) {
+	// ginkgo can run tests in parallel. However, we really just need this mapping to be
+	// performed once for the whole duration of a test.
+	// Therefore we store the result in a global variable, and then lock
+	// the access to it.
+	lock.Lock()
+	defer lock.Unlock()
+	// if mapping has already been done, return it right away
+	if mapping != nil {
+		return mapping, nil
+	}
+	// get compatible versions for subnetEVM
+	// subnetEVMversions is a list of sorted EVM versions,
+	// subnetEVMmapping maps EVM versions to their RPC versions
+	subnetEVMversions, subnetEVMmapping, err := getVersions(mapper, models.SubnetEvm)
+	if err != nil {
+		return nil, err
+	}
+	// subnet-evm publishes its upcoming new version in the compatibility json
+	// before the new version is actually a downloadable release
+	subnetEVMversions = subnetEVMversions[1:]
```

Comment:
OK. 

For avalanchego, the code will only download versions based on a RPC which we get from a VM - which means it will always download a version which has already been released.

And for the tests where avalanchego runs alone, we just download `latest`. So for this use case, I don't need to apply that thing to avalanchego.